Mathlib/Data/SetLike/Basic.lean

```diff
@@ -243,4 +243,13 @@ lemma mem_of_subsingleton {A F} [Subsingleton A] [SetLike F A] (S : F) [h : None
   obtain ⟨s, hs⟩ := nonempty_subtype.mp h
   simpa [Subsingleton.elim a s]
 
+variable {A : Type*} {S : Type*} [SetLike S A] [PartialOrder S] [OrderTop S]
+
+/-- If `s` is a proper element of a `SetLike` structure (i.e., `s ≠ ⊤`) and the top element
+coerces to the universal set, then there exists an element not in `s`. -/
+lemma exists_not_mem_of_ne_top (s : S) (hs : s ≠ ⊤) (h_top : ((⊤ : S) : Set A) = Set.univ) :
+    ∃ a : A, a ∉ s := by
+  have : (s : Set A) ≠ Set.univ := fun h_eq => hs (coe_injective (h_eq.trans h_top.symm))
+  exact (Set.ne_univ_iff_exists_notMem _).mp this
```

Comment:
I'm not sure how useful this is in practice, but to make it more so, you can let `h_top` have an autoparam.
```
lemma exists_not_mem_of_ne_top (s : S) (hs : s ≠ ⊤)
    (h_top : ((⊤ : S) : Set A) = Set.univ := by simp) :
    ∃ a : A, a ∉ s := by
  simpa [-SetLike.coe_set_eq, SetLike.ext'_iff, h_top, Set.ne_univ_iff_exists_notMem] using hs
```

Reply:
Yeah, this is admittedly a minor contribution but I'm working on a couple theorems that have an existential form and I found this one to be useful.